src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.NativeAot.cs

```diff
@@ -471,15 +480,76 @@ public ManagedObjectWrapperReleaser(ManagedObjectWrapper* wrapper)
         internal unsafe class NativeObjectWrapper
         {
             internal IntPtr _externalComObject;
+            private IntPtr _inner;
             private ComWrappers _comWrappers;
             internal GCHandle _proxyHandle;
+            internal GCHandle _proxyHandleTrackingResurrection;
+            internal GCHandle _nativeObjectWrapperWeakHandle;
+
+            private readonly bool _aggregatedManagedObjectWrapper;
+            internal readonly bool _fromTrackerRuntime;
+            private IntPtr _trackerObject;
+            private readonly bool _releaseTrackerObject;
+            private int _trackerObjectDisconnected;
+
+            internal readonly IntPtr _contextToken;
 
-            public NativeObjectWrapper(IntPtr externalComObject, ComWrappers comWrappers, object comProxy)
+            public IntPtr TrackerObject => (_trackerObject == IntPtr.Zero || _trackerObjectDisconnected == 1) ? IntPtr.Zero : _trackerObject;
+
+            public NativeObjectWrapper(IntPtr externalComObject, IntPtr inner, ComWrappers comWrappers, object comProxy, CreateObjectFlags flags)
```

Comment:
Would it be interesting to split the aggregation and tracker runtime support into separate subclasses of `NativeObjectWrapper` or introduce a base class/interface?

It might help make the code cleaner/easier to understand, especially since we have the `CreateObjectFlags` data present at the time we create this wrapper.

This work can also wait until after merging.

Reply:
Just curious, would this also lead to some memory usage improvements for the non-aggregation/tracker scenario? Since it seems there's quite a few additional fields being needed for those, which also seem to be per-object? 🤔

Reply:
I agree. If this is kept as-is, we should understand whether the perf regression for plain vanilla COM interop is acceptable before merging.

Reply:
Addressed by creating a new subclass for the tracker scenarios.  Left the aggregation scenario as there is only one bool it uses.